### PR TITLE
45-Create end point to update trainees details -Olena

### PR DESCRIPTION
**Pull request by OlenaReukova**
in traineeController.js
line #8 const { traineeId } = req.params;
may need to be replaced by
const { id} = req.params;

line #12
const result = await db.query(q, [traineeId]);
may need to be replaced  by
 const result = await db.query(q, [id]);

endpoint updates trainees' details  according to the table schema (id, gitHub_user_name, full_name, cohort_id)

### CODE REVIEW

**Comment by OlenaReukova** [comment]
@PeteLindsell review please

File: server/controllers/traineeController.js [hunk 1]
```diff
@@ -1,48 +1,63 @@
 import db from "../db";
 
 export const getAllTrainees = (req, res) => {
-    res.status(200).json("hello from trainees route");
+	res.status(200).json("hello from trainees route");
 };
 
 export const getTrainee = async (req, res) => {
-    const { traineeId } = req.params;
```
Comment (Zobeir-Rigi) [inline server/controllers/traineeController.js:8]: we don't have traineeId in the table, its only id

File: server/controllers/traineeController.js [hunk 2]
```diff
@@ -1,48 +1,63 @@
...
+	try {
+		const q =
+			"UPDATE trainees SET full_name = $1, gitHub_user_name= $2, cohort_id = $3 WHERE id = $4";
+		await db.query(q, [full_name, cohort_id, id, gitHub_user_name]);
+		res.status(200).json({ message: "Trainee details updated successfully" });
+	} catch (error) {
+		console.error("Error updating trainee details:", error);
+		res.status(500).send("Failed, Internal Server Error");
+	}
+};
```
Comment (Zobeir-Rigi) [inline server/controllers/traineeController.js:63]: good code

File: server/controllers/traineeController.js (hunk 1 above)
Comment (OlenaReukova) [reply server/controllers/traineeController.js:8]: @Zobeir-Rigi yes you are right, we discussed it with Boshra, I should change it.

File: server/controllers/traineeController.js (hunk 2 above)
Comment (OlenaReukova) [reply server/controllers/traineeController.js:63]: @Zobeir-Rigi thank you

File: server/controllers/traineeController.js (hunk 1 above)
Comment (OlenaReukova) [reply server/controllers/traineeController.js:8]: @Zobeir-Rigi yes you are right, we discussed it with Boshra, I should change it.

Comment (BoshraM) [reply server/controllers/traineeController.js:8]: @OlenaReukova  yes you change it to id , just one point when you equal something with req.params its not related to data base , this is something regarding your endpoint and request.

**Review by PeteLindsell — APPROVED** [review]
Looks good to me 👍

**Pull request merged**